solr/contrib/extraction/src/java/org/apache/solr/handler/extraction/XLSXResponseWriter.java

```diff
@@ -357,7 +357,7 @@ public void writeArray(String name, Iterator val) throws IOException {
       if (v instanceof IndexableField) {
         IndexableField f = (IndexableField)v;
         if (v instanceof Date) {
-          output.append(((Date) val).toInstant().toString() + "; ");
+          output.append(((Date) v).toInstant().toString() + "; ");
```

Comment:
This looks legitimate - Would cause a classCastException. But have we ever seen it in the wild?

Reply:
I have not seen this in the logs.

I hit some weird error a few years back and I couldn’t believe how that we have such blatant bugs in the code all over that I ran findbugs and started wack-a-mole game for all the bugs reported. 

I don’t remember what the original bug was anymore. I will give this a second shot and I will put small patches for each type of error reported by findbugs